src/vs/workbench/contrib/terminal/browser/xterm/decorationAddon.ts

```diff
@@ -117,12 +117,14 @@ export class DecorationAddon extends Disposable implements ITerminalAddon {
 		this.clearDecorations();
 	}
 
+	private _clearPlaceholder(): void {
+		this._placeholderDecoration?.dispose();
+		this._placeholderDecoration = undefined;
+	}
+
 	public clearDecorations(): void {
-		if (this._placeholderDecoration) {
-			this._placeholderDecoration.marker.dispose();
-			this._placeholderDecoration.dispose();
-			this._placeholderDecoration = undefined;
```

Comment:
line 123 was getting reached when `_placeholderDecoration` had already gotten set to undefined via the `onDispose` below